db/warehouse/migrate/20240731144633_add_verified_by_project_id_to_current_living_situation.rb

```diff
@@ -0,0 +1,6 @@
+class AddVerifiedByProjectIdToCurrentLivingSituation < ActiveRecord::Migration[7.0]
+  def change
+    add_column :CurrentLivingSituation, :verified_by_project_id, :integer
+    add_foreign_key :CurrentLivingSituation, :Project, column: :verified_by_project_id, validate: false
```

Comment:
related to my comment below about `strong_migrations,` this also results in a similar warning:
```
=== Dangerous operation detected #strong_migrations ===

Adding a foreign key blocks writes on both tables. Instead, use:

class AddVerifiedByProjectIdToCurrentLivingSituation < ActiveRecord::Migration[7.0]
  disable_ddl_transaction!

  def change
    add_reference :CurrentLivingSituation, :verified_by_project, index: {algorithm: :concurrently}, null: true
  end
end

Then add the foreign key in separate migrations.

/app/db/warehouse/migrate/20240731144633_add_verified_by_project_id_to_current_living_situation.rb:3:in `change'
```

It looks like one practice we use is to just use `safety_assured` or `disable_check` with `add_reference`, so I would be fine with doing that here if that's the preferred method.

Reply:
Makes sense, but I still don't feel that confident in my understanding. From what I read, the reason to prefer `add_reference` over `add_foreign_key` is because it's a shortcut for adding the column and the foreign key at the same time, which is what we want to do here; and because it's more common in our codebase as Theron pointed out in the initial comment on this thread. But to avoid blocking writes on both the Project and CLS table at the same time while adding the foreign key, StrongMigrations recommends adding the foreign key separately anyway; in that case, is there any advantage to using `add_reference` at all? Or should I revert this to what I had before?

Reply:
Resolved on a call - we will use `safety_assured` (ignore StrongMigrations) here because validating a column full of nulls should still be relatively fast.